File: leetcode/easy/575-distributeCandies.java
// first pass solution
class Solution {
    public int distributeCandies(int[] candyType) {
        Set<Integer> set = new HashSet<>();
        int max = candyType.length / 2;
        
        for (int n : candyType) {
            set.add(n);
            if (set.size() >= max) return max;
        }
        return set.size();
    }
}

File: leetcode/easy/575-distributeCandies.cpp
class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        int half = (int) candyType.size() / 2;
        set<int> set;

        for (const auto candy: candyType) {
            set.insert(candy);
        }
        return min<int>(half, set.size());
    }
};
